Clang release build error

Building with clang 8.0.1 on Ubuntu 18.04 produces the following error:
```
OpenCL-CTS/test_conformance/spir/run_services.cpp:317:1: error: control may reach end of non-void function [-Werror,-Wreturn-type]
}
^
1 error generated.
```
PR is in progress.

--- a/test_conformance/spir/run_services.cpp
+++ b/test_conformance/spir/run_services.cpp
@@ -302,18 +302,11 @@ const std::string& DataRow::operator[](int column)const
 
 std::string& DataRow::operator[](int column)
 {
-    assert((column > -1) && "Index out of bound");
+    assert((column > -1 && (size_t)column <= m_row.size())
+           && "Index out of bound");
+    if ((size_t)column == m_row.size()) m_row.push_back("");
 
-    if ((size_t)column < m_row.size())
-        return m_row[column];
-
-    if (column == m_row.size())
-    {
-        m_row.push_back("");
-        return m_row[column];
-    }
-
-    assert(0 && "Index out of bound.");
+    return m_row[column];
 }
 
 /*

